security/pkg/nodeagent/sds/sdsservice.go

```diff
@@ -156,8 +184,33 @@ func (s *sdsservice) FetchSecrets(ctx context.Context, discReq *xdsapi.Discovery
 	return sdsDiscoveryResponse(secret, proxy)
 }
 
-func (s *sdsservice) pushSDS(secret *cache.SecretItem, proxy model.Proxy, con *sdsConnection) error {
-	response, err := sdsDiscoveryResponse(secret, proxy)
+// NotifyProxy send notification to proxy about secret update.
```

Comment:
s/send/sends

Can you document "If `secret` is nil, the stream will be killed.`?

Reply:
added comments